modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/OrphanDetector.java

```diff
@@ -195,7 +182,8 @@ private void checkTxOrphanedInternal(UUID txId) {
             sentTxRecoveryMessage(txState.commitPartitionId(), txId);
         }
 
-        throw new TransactionException(ACQUIRE_LOCK_ERR, "The lock is held by the abandoned transaction [abandonedTxId=" + txId + "].");
+        return failedFuture(
+                new TransactionException(ACQUIRE_LOCK_ERR, "The lock is held by the abandoned transaction [abandonedTxId=" + txId + "]."));
```

Comment:
I don't think it is definitely right to ignore the exception. Let's add TODO here.